src/compiler/checker.ts

```diff
@@ -14031,6 +14031,23 @@ namespace ts {
             }
         }
 
+        /** Check that type parameter lists are identical across multiple declarations */
+        function checkTypeParameterListsIdentical(node: ClassLikeDeclaration | InterfaceDeclaration, symbol: Symbol) {
+            let firstDecl: ClassLikeDeclaration | InterfaceDeclaration;
+            if (symbol.declarations.length > 1) {

```

Comment:
The second simplification is not safe because firstDecl needs to be `ClassLikeDecl | InterfaceDecl`, which might not be true if there's a module in there too.